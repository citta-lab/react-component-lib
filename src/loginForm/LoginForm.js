import React from 'react';

import Input from './Input';
import Button from '../button/Button';

const LoginForm = () => (
  <form>
    <Input
      className="form-username"
      value=""
      type="text"
      name="username"
      placeholder="Enter your username"
    />
    <Input
      className="form-password"
      value=""
      type="password"
      name="username"
      placeholder="Enter your password"
    />
    <Button type="submit" text="Press" />
  </form>
);

export default LoginForm;
