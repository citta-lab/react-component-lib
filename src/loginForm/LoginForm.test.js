import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import LoginForm from './LoginForm';

Enzyme.configure({ adapter: new Adapter() });

it('converts basic button using shallow', () => {
  const shallowed = shallow(<LoginForm />);
  expect(shallowed).toMatchSnapshot();
});

it('converts basic button using mount', () => {
  const mounted = mount(<LoginForm />);
  expect(mounted).toMatchSnapshot();
});

it('converts basic button using render', () => {
  const rendered = render(<LoginForm />);
  expect(rendered).toMatchSnapshot();
});
