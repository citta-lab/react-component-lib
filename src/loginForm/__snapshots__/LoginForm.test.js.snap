// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts basic button using mount 1`] = `
<LoginForm>
  <form>
    <Input
      className="form-username"
      name="username"
      placeholder="Enter your username"
      type="text"
      value=""
    >
      <input
        className="form-username"
        name="username"
        placeholder="Enter your username"
        type="text"
        value=""
      />
    </Input>
    <Input
      className="form-password"
      name="username"
      placeholder="Enter your password"
      type="password"
      value=""
    >
      <input
        className="form-password"
        name="username"
        placeholder="Enter your password"
        type="password"
        value=""
      />
    </Input>
    <Button
      text="Press"
      type="submit"
    >
      <button
        type="submit"
      >
        Press
      </button>
    </Button>
  </form>
</LoginForm>
`;

exports[`converts basic button using render 1`] = `
<form>
  <input
    class="form-username"
    name="username"
    placeholder="Enter your username"
    type="text"
    value=""
  />
  <input
    class="form-password"
    name="username"
    placeholder="Enter your password"
    type="password"
    value=""
  />
  <button
    type="submit"
  >
    Press
  </button>
</form>
`;

exports[`converts basic button using shallow 1`] = `
<form>
  <Input
    className="form-username"
    name="username"
    placeholder="Enter your username"
    type="text"
    value=""
  />
  <Input
    className="form-password"
    name="username"
    placeholder="Enter your password"
    type="password"
    value=""
  />
  <Button
    text="Press"
    type="submit"
  />
</form>
`;
